### PR TITLE
test(dataset): finish dataset test

### PR DESCRIPTION
**Pull request by qubaomingg**
### 🤔 PR includes

✨ Feature

- [ ] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve performance.
- [ ] Type optimization

🐛 Bug fix

- [ ] Bug fix issue #0

🔧 Chore

- [x] repair all test case
- [x] unified test data & option
- [x] finish data-set test

### 🖼️ Screenshot

|  Before  |  After  |
|----|----|
|  ❌  |  ✅  |

### CODE REVIEW

File: packages/s2-core/__tests__/unit/core/data-process/pivot-spec.tsx [hunk 1]
```diff
@@ -1,350 +1,344 @@
...
-      expect(getCellMeta(7, 1).data[VALUE_FIELD]).toBe(126);
-      expect(getCellMeta(6, 0).data[VALUE_FIELD]).toBe(273);
-      expect(getCellMeta(6, 1).data[VALUE_FIELD]).toBe(273);
-    });
-  });
-});
+ import { flattenDeep, get, size, uniq } from 'lodash';
+ import { assembleDataCfg, assembleOptions } from '../../../util/sheet-entry';
+ import { getContainer } from '../../../util/helpers';
+ import { data } from '../../../data/mock-dataset.json';
```
Comment (wjgogogo) [inline packages/s2-core/__tests__/unit/core/data-process/pivot-spec.tsx:7]: 统一为`@`

File: packages/s2-core/__tests__/unit/core/data-process/table-spec.tsx [hunk 2]
```diff
@@ -4,94 +4,91 @@
...
-      expect(getCellMeta(2, 1).data).toEqual({ city: '宁波市' });
-      expect(getCellMeta(2, 2).data).toEqual({ category: '家具' });
-      expect(getCellMeta(2, 3).data).toEqual({ subCategory: '桌子' });
-      expect(getCellMeta(2, 4).data).toEqual({ price: 273 });
-    });
-  });
-});
+ import { get } from 'lodash';
+ import { SpreadSheet } from 'src/sheet-type';
+ import { assembleDataCfg, assembleOptions } from '../../../util/sheet-entry';
```
Comment (wjgogogo) [inline packages/s2-core/__tests__/unit/core/data-process/table-spec.tsx:9]: 同上

File: packages/s2-core/__tests__/unit/dataset/pivot-dataset-spec.ts [hunk 3]
```diff
@@ -1,337 +1,231 @@
...
+     test('getCellData function', () => {
+       expect(dataSet.getCellData({ 
+         query: {
+           province: '浙江省',
+           city: '杭州市',
+           type: '家具',
+           sub_type: '桌子',
+           [EXTRA_FIELD]: 'price'
+         }
+       })).toEqual(expect.objectContaining({
```
Comment (wjgogogo) [inline packages/s2-core/__tests__/unit/dataset/pivot-dataset-spec.ts:125]: 可以试试这个断言：https://github.com/jest-community/jest-extended#tocontainentrieskey-value

File: packages/s2-core/__tests__/unit/core/data-process/table-spec.tsx [hunk 4]
```diff
@@ -4,94 +4,91 @@
...
+       expect(getCellMeta(0, 4).data).toEqual({ price: 1 });
+       // 第三行
+       expect(getCellMeta(2, 0).data).toEqual({ province: '浙江省' });
+       expect(getCellMeta(2, 1).data).toEqual({ city: '宁波市' });
+       expect(getCellMeta(2, 2).data).toEqual({ type: '家具' });
+       expect(getCellMeta(2, 3).data).toEqual({ sub_type: '桌子' });
+       expect(getCellMeta(2, 4).data).toEqual({ price: 3 });
+     });
+   });
+ });
```
Comment (xingwanying) [inline packages/s2-core/__tests__/unit/core/data-process/table-spec.tsx:94]: image

File: packages/s2-core/__tests__/unit/data-accuracy-two-measures-spec.tsx [hunk 5]
```diff
@@ -240,11 +240,10 @@ describe('data accuracy two measures spec', () => {
   act(() => {
     ReactDOM.render(<MainLayout />, getContainer());
   });
-
   spreadsheet1.setDataCfg(getDataCfg(6));
   test('Totals + Details + Tow Measures', () => {
     expect(data6.length).toBe(8);
-    expect(spreadsheet1.dataSet.originData.length).toBe(16);
+    expect(spreadsheet1.dataSet.originData.length).toBe(8);
```
Comment (xingwanying) [inline packages/s2-core/__tests__/unit/data-accuracy-two-measures-spec.tsx:246]: 这个地方好迷啊……一会儿8一会儿16的，我都晕了……

File: packages/s2-core/__tests__/unit/dataset/pivot-dataset-row-value-spec.ts [hunk 6]
```diff
@@ -1,10 +1,237 @@
...
+             province: '浙江省',
+             city: '杭州市',
+           }
+         },
+       ]
+       dataSet.setDataCfg(dataCfg);
+       expect(dataSet.getDimensionValues('sub_type')).toEqual(['桌子', '沙发', '笔', '纸张' ]);
+     });
+   });
+ });
```
Comment (xingwanying) [inline packages/s2-core/__tests__/unit/dataset/pivot-dataset-row-value-spec.ts:237]: 这里也是差个空行

File: packages/s2-core/__tests__/unit/dataset/table-dataset-spec.ts [hunk 7]
```diff
@@ -1,10 +1,67 @@
...
+ 
+       expect(dataSet.getCellData({ 
+         query: {
+           rowIndex: 5,
+           col: 'sub_type'
+         }
+       })).toEqual('沙发');
+     });
+   });
+ });
```
Comment (xingwanying) [inline packages/s2-core/__tests__/unit/dataset/table-dataset-spec.ts:67]: 同上

File: packages/s2-core/__tests__/unit/utils/split-total-spec.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+       rows: ['province', 'city'],
+       columns: ['category', 'subCategory'],
+     };
+     const totals = splitTotal([].concat(data).concat(totalData), fields);
+     totals.forEach((total) => {
+       const dimensions = [].concat(fields.rows).concat(fields.columns);
+       expect(every(dimensions, (dimension) => total[dimension])).toBe(false);
+     });
+   });
+ });
```
Comment (xingwanying) [inline packages/s2-core/__tests__/unit/utils/split-total-spec.ts:20]: 同上

File: packages/s2-core/__tests__/util/sheet-entry.tsx [hunk 9]
```diff
@@ -151,4 +153,4 @@ export const SheetEntry = forwardRef(
       </div>
     );
   },
-);
+);
```
Comment (xingwanying) [inline packages/s2-core/__tests__/util/sheet-entry.tsx:156]: 同上

File: packages/s2-core/__tests__/unit/core/data-process/pivot-spec.tsx (hunk 1 above)
Comment (qubaomingg) [reply packages/s2-core/__tests__/unit/core/data-process/pivot-spec.tsx:7]: src/的可是用@，__tests__ 文件夹的如何用呢

File: packages/s2-core/__tests__/unit/data-accuracy-two-measures-spec.tsx (hunk 5 above)
Comment (qubaomingg) [reply packages/s2-core/__tests__/unit/data-accuracy-two-measures-spec.tsx:246]: 我觉得应该是[]。。。算了注释掉吧，这个单测还要改

**Pull request merged**